vimiv/imutils/exif.py

```diff
@@ -261,13 +293,95 @@ def copy_exif(self, dest: str, reset_orientation: bool = True) -> None:
         except OSError as e:
             _logger.debug("Failed to write exif data for '%s': '%s'", dest, str(e))
 
-    def exif_date_time(self) -> str:
+    def get_date_time(self) -> str:
         with contextlib.suppress(KeyError):
             return self._metadata["Exif.Image.DateTime"].raw_value
         return ""
 
 
-has_exif_support = ExifHandler != _ExifHandlerBase
+has_exif_support = _ExternalKeyHandler != _ExternalKeyHandlerBase
+
+
+class MetadataHandler:
+    """Handler to load and copy exif information of a single image.
+
+    This class provides the interface for handling exif support. By default none of the
+    operations are implemented. Instead it is up to a child class which wraps around one
+    of the supported exif libraries to implement the methods it can.
+    """
+
+    def __init__(self, filename=""):
+        self.filename = filename
+        self._internal_handler: Optional[_InternalKeyHandler] = None
+        self._external_handler: Optional[_ExternalKeyHandler] = None
+
+    @property
+    def internal_handler(self) -> _InternalKeyHandler:
+        if self._internal_handler is None:
+            self._internal_handler = _InternalKeyHandler(self.filename)
+        return self._internal_handler
+
+    @property
+    def external_handler(self) -> _ExternalKeyHandler:
+        if self._external_handler is None:
+            self._external_handler = _ExternalKeyHandler(self.filename)
+        return self._external_handler
+
+    def copy_metadata(self, dest: str, reset_orientation: bool = True) -> None:
+        """Copy exif information from current image to dest.
+
+        Args:
+            dest: Path to write the exif information to.
+            reset_orientation: If true, reset the exif orientation tag to normal.
+        """
+        try:
+            self.external_handler.copy(dest, reset_orientation)
+        except UnsupportedExifOperation:
```

Comment:
If we decide to keep these methods in the metadata handler, I would certainly let the exception propagate. But not having them here at all, seems even nicer to me :blush:

Reply:
I have removed them in d3281b46867badf1db503c8be7375504cbfc9b59